Add span queries

alekitto opened the pull request:
Fixes #1316 

- [x] span_term
- [x] span_multi
- [x] span_first
- [x] span_near
- [x] span_or
- [x] span_containing
- [x] span_within
- [x] span_not
- [ ] field_masking_span

ruflin commented:
https://github.com/ruflin/Elastica/pull/1320 is now merged. This probably needs rebase now.

alekitto commented:
I've rebased and squashed my work.
The only thing left is the `field_masking_span`, but it's barely documented (there's only an example), so i think we could leave this out, as the implementation of the query class would be probably wrong.

ruflin commented on lib/Elastica/Query/SpanContaining.php:
```diff
@@ -0,0 +1,45 @@
+     */
+    public function setLittle(AbstractSpanQuery $little)
+    {
+        $this->setParam('little', $little);
```
We normal have a `return $this->setParam('little', $little);` in here so all setters return the object itself. This allows for chaining functions.

alekitto commented:
Ok, fixed all the setters in SpanContaining, SpanNot, SpanWithin and SpanNear (missed in #1320)

ruflin commented:
@alekitto In the diff Builder.php shows up and I'm not sure why?

ruflin commented:
BTW: Could you also add an entry to the changelog?

alekitto commented:
Builder.php is an empty file, I probably deleted it without noticing.
The Changelog already contains a line for Span* queries addition. Do I have to add another one for these three?

ruflin commented:
Argh, it seems some things got mixed up in the Changelog :-( The reason an additional entry is needed because just yesterday happened a release. So this goes into the next release.

ruflin commented:
I just pushed a change to the CHANGELOG to bring it up-to-date. Make sure to get the most recent one in before updating it.

alekitto commented:
👍 Done.
